tests/jax/components/training/step_test.py

```diff
@@ -0,0 +1,723 @@
+
+
+def test_mapg_with_trust_region_step_initiator() -> None:
+    """Test constructor of MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    assert mapg_with_trust_region_step.config.discount == 0.99
+
+
+def test_on_training_init_start(mock_trainer: MockTrainer) -> None:
+    """Test on_training_init_start method from MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    mapg_with_trust_region_step.on_training_init_start(trainer=mock_trainer)
+
+    assert (
+        mock_trainer.store.full_batch_size
+        == mock_trainer.store.sample_batch_size
+        * (mock_trainer.store.sequence_length - 1)
+    )
+
+
+def test_on_training_step_fn(mock_trainer: MockTrainer) -> None:
+    """Test on_training_step_fn method from MAPGWITHTrustRegionStep component"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    del mock_trainer.store.step_fn
+    mapg_with_trust_region_step.on_training_step_fn(trainer=mock_trainer)
+
+    assert callable(mock_trainer.store.step_fn)
+
+
+def test_step(mock_trainer: MockTrainer, dummy_sample: DummySample) -> None:
+    """Test step function"""
+    mapg_with_trust_region_step = MAPGWithTrustRegionStep()
+    del mock_trainer.store.step_fn
+    mapg_with_trust_region_step.on_training_step_fn(trainer=mock_trainer)
+    old_key = mock_trainer.store.key
+    with jax.disable_jit():
+        metrics = mock_trainer.store.step_fn(dummy_sample)
+
+    assert list(metrics.keys()) == [
+        "norm_params",
+        "observations_mean",
+        "observations_std",
+        "rewards_mean",
+        "rewards_std",
+    ]
+
+    updates = {
+        net_key: mock_trainer.store.networks["networks"][net_key].params
+        for net_key in mock_trainer.store.networks["networks"].keys()
+    }
+
+    assert metrics["norm_params"] == optax.global_norm(jax.tree_leaves(updates))
+
+    assert metrics["observations_mean"] == 0.5667871
+
+    assert metrics["observations_std"] == 0.104980744
+
+    assert list(metrics["rewards_mean"].keys()) == ["agent_0", "agent_1", "agent_2"]
+    assert float(list(metrics["rewards_mean"].values())[0]) == 0.0
+    assert float(list(metrics["rewards_mean"].values())[1]) == 0.08261970430612564
+    assert float(list(metrics["rewards_mean"].values())[2]) == 0.07156813144683838
+
+    assert list(metrics["rewards_std"].keys()) == ["agent_0", "agent_1", "agent_2"]
+    assert float(list(metrics["rewards_std"].values())[0]) == 0.0
+    assert float(list(metrics["rewards_std"].values())[1]) == 0.07048596441745758
+    assert float(list(metrics["rewards_std"].values())[2]) == 0.07740379124879837
+
+    random_key, _ = jax.random.split(old_key)
+    assert list(mock_trainer.store.key) == list(random_key)
+
+    assert mock_trainer.store.opt_states == {
+        "network_agent_0": 0,
+        "network_agent_1": 1,
+        "network_agent_2": 2,
+    }
```

Comment:
I know this would be quite a bit of work, but an important part of the step component is that the network parameters and the optimizer states get updated/changed. I think a way we could do this is to update the mock epoch update function to not return the same carry as was passed in but updating the parameters and states. Even if we just increment everything by 1 each time the epoch update is called. That way we can verify that parameters and states are being update and that the epoch update funciton is getting called the correct number of times.